app/models/concerns/image_file.rb

```diff
@@ -110,14 +110,18 @@ def reset_image_attributes
     self.status = :created
   end
 
+  # only replace if there is a new image url, different from current
   def replace?(img)
-    original_url != img.try(:original_url)
+    original_url && (original_url != img.try(:original_url))
   end
 
   def update_image(img)
     %i[alt_text caption credit].each do |key|
       img[key] = self[key]
     end
+
+    # autosave won't work; explictly save, as img is image is detached from the association
+    img.save
```

Comment:
Hrm... why does autosave not happen?  It's doing a `img.update_image(image)`, and that image is `images.first` ... and `has_many :images` has autosave on it.

I wonder if this could cause issues with the RSS importer code (it also uses this setter in several places).

Reply:
rails AR autosave does insert and delete, not update

Reply:
Hrm, this appears to do the update for me.  In the console:

```ruby
e = EpisodeImage.last.episode
e.images[0].caption = "whatev"
e.save!
```

Reply:
you can see it's still an issue https://github.com/rails/rails/issues/17466

Reply:
I believe the call to `images.first` gets the object, which is not connected to the association anymore

Reply:
for example, this does not work:
```
e = EpisodeImage.last.episode
i = e.image
i.alt_text = "er"
e.save!
```

Reply:
Oh, I see.  Should we be doing `images[0]` then?  It seems like in our code, there's never going to be more than 1 (due to the `replace_resources!` hook deleting the others).